vcd/resource_vcd_snat.go

```diff
@@ -93,7 +107,7 @@ func resourceVcdSNATRead(d *schema.ResourceData, meta interface{}) error {
 	var found bool
 
 	for _, r := range e.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule {
-		if r.RuleType == "SNAT" &&
+		if r.RuleType == "SNAT" && r.GatewayNatRule.Interface.Name == d.Get("network_name") &&
```

Comment:
But the `network_name` is optional for now. It will start skipping this for who is not using new functionality.

Reply:
Addressed